### ISSUE TITLE
Warn on unmatched "nolint" directives

### ISSUE THREAD

**Issue by leonklingele** [golangci/golangci-lint#404]
Please add a `--warn-unmatched-nolint` flag (which is supported by [gometalinter](https://github.com/alecthomas/gometalinter/blob/3277e059b28c81daaa3d014ca2afec7d3af3651f/main.go#L73)) which warns on unmatched "nolint" directives. Example:

```go
func ThisWillNeverFail() {
    println("Hello") // nolint: lll
}
```

`nolint: lll` in this case will never catch any issues and therefore should be removed.

**Comment by jirfag** [comment]
Hi, It's a good idea, thank you. The question is do we need to exit with error code if these cases are found

**Comment by leonklingele** [comment]
> The question is do we need to exit with error code if these cases are found

Yeah, why not? A warning could be issued by default and an exit error code could be used if `--treat-warnings-as-errors` (or something similar) is passed.

**Comment by leonklingele** [comment]
Didn't mean to close.

**Comment by leonklingele** [comment]
No

**Comment by ldez** [comment]
`nolintlint` handles those cases now.

```yml
  nolintlint:
    # Enable to ensure that nolint directives are all used. Default is true.
    allow-unused: false
    # Disable to ensure that nolint directives don't have a leading space. Default is true.
    allow-leading-space: true
    # Exclude following linters from requiring an explanation.  Default is [].
    allow-no-explanation: []
    # Enable to require an explanation of nonzero length after each nolint directive. Default is false.
    require-explanation: true
    # Enable to require nolint directives to mention the specific linter being suppressed. Default is false.
    require-specific: true
```
